pyrokinetics/gk_code/GKInputGS2.py

```diff
@@ -212,6 +212,99 @@ def get_local_species(self):
         local_species.normalise()
         return local_species
 
+    def _read_single_grid(self):
+        return {
+            "nky": 1,
+            "nkx": 1,
+            "ky": self.data["kt_grids_single_parameters"]["aky"] / sqrt2,
+            "kx": 0.0,
+            "theta0": self.data["kt_grids_single_parameters"].get("theta0", 0.0),
+        }
+
+    def _read_range_grid(self):
+        range_options = self.data["kt_grids_range_parameters"]
+        nky = range_options.get("naky", 1)
+
+        ky_min = range_options.get("aky_min", "0.0")
+        ky_max = range_options.get("aky_max", "0.0")
+
+        spacing_option = range_options.get("kyspacing_option", "linear")
+        if spacing_option == "default":
+            spacing_option = "linear"
+
+        ky_space = np.linspace if spacing_option == "linear" else np.logspace
+
+        ky = ky_space(ky_min, ky_max, nky) / sqrt2
+
+        return {
+            "nky": nky,
+            "nkx": 1,
+            "ky": ky,
+            "kx": np.array([0.0]),
+            "theta0": 0.0,
+        }
+
+    def _read_box_grid(self):
+        box = self.data["kt_grids_box_parameters"]
+        keys = box.keys()
+
+        grid_data = {}
+
+        # Set up ky grid
+        if "ny" in keys:
+            grid_data["nky"] = int((box["ny"] - 1) / 3 + 1)
+        elif "n0" in keys:
```

Comment:
`n0` is not related to the number of modes, but sets the toroidal mode number corresponding to the first non-zero ky. The number of ky should always be set by ny or naky here I think.

Reply:
Yes, `n0` can be used to set `ky` via `ky=n0*drhodpsi*rhostar_single`